File: Easy/tassaiscrizioneconf/meta_tassaiscrizioneconfdefaultview/meta_tassaiscrizioneconfdefaultview.cs
/*
    Easy
    Copyright (C) 2020 Università degli Studi di Catania (www.unict.it)
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

﻿using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_tassaiscrizioneconfdefaultview
{
    public class Meta_tassaiscrizioneconfdefaultview : Meta_easydata 
	{
        public Meta_tassaiscrizioneconfdefaultview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "tassaiscrizioneconfdefaultview") {
				Name = "Definizione delle tasse di iscrizione";
			EditTypes.Add("default");
			ListingTypes.Add("default");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idtassaiscrizioneconf"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "default": {
						DescribeAColumn(T, "tassaiscrizioneconf_aa", "Anno accademico", nPos++);
						DescribeAColumn(T, "title", "Title", nPos++);
						DescribeAColumn(T, "tassaiscrizioneconf_aamax", "Anno accademico massimo", nPos++);
						DescribeAColumn(T, "tassaiscrizioneconf_aamin", "Anno accademico minimo", nPos++);
						DescribeAColumn(T, "tassaiscrizioneconf_annofcmax", "Anno di iscrizione fuori corso massimo", nPos++);
						DescribeAColumn(T, "tassaiscrizioneconf_annofcmin", "Anno di iscrizione fuori corso minimo", nPos++);
						DescribeAColumn(T, "tassaiscrizioneconf_annomax", "Anno di iscrizione massimo", nPos++);
						DescribeAColumn(T, "tassaiscrizioneconf_annomin", "Anno di iscrizione minimo", nPos++);
						DescribeAColumn(T, "tassaiscrizioneconf_codice_corsostudio", "Codice del corso di studio", nPos++);
						DescribeAColumn(T, "tassaiscrizioneconf_codice_didprog", "Codice della didattica programmata", nPos++);
						DescribeAColumn(T, "tassaiscrizioneconf_codice_didprogcurr", "Codice del curriculum", nPos++);
						DescribeAColumn(T, "tassaiscrizioneconf_codice_didprogori", "Codice dell'orientamento", nPos++);
						DescribeAColumn(T, "tassaiscrizioneconf_corsisingoli", "Corsi singoli", nPos++);
						DescribeAColumn(T, "costoscontodef_title", "Costo", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		public override string GetSorting(string ListingType) {
			switch (ListingType) {
				case "default": {
						return "title asc ";
					}
				//$GetSorting$
			}
			return base.GetSorting(ListingType);
		}

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/tassaiscrizioneconf/meta_tassaiscrizioneconfdefaultview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_tassaiscrizioneconfdefaultview() {
        MetaData.apply(this, ["tassaiscrizioneconfdefaultview"]);
        this.name = 'meta_tassaiscrizioneconfdefaultview';
    }

    meta_tassaiscrizioneconfdefaultview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_tassaiscrizioneconfdefaultview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'default':
						this.describeAColumn(table, 'title', 'Titolo', null, 3000, 2024);
						this.describeAColumn(table, 'aamax', 'Anno accademico massimo', null, 4000, 9);
						this.describeAColumn(table, 'aamin', 'Anno accademico minimo', null, 5000, 9);
						this.describeAColumn(table, 'tassaiscrizioneconf_annofcmax', 'Anno di iscrizione fuori corso massimo', null, 6000, null);
						this.describeAColumn(table, 'tassaiscrizioneconf_annofcmin', 'Anno di iscrizione fuori corso minimo', null, 7000, null);
						this.describeAColumn(table, 'tassaiscrizioneconf_annomax', 'Anno di iscrizione massimo', null, 8000, null);
						this.describeAColumn(table, 'tassaiscrizioneconf_annomin', 'Anno di iscrizione minimo', null, 9000, null);
						this.describeAColumn(table, 'tassaiscrizioneconf_codice_corsostudio', 'Codice del corso di studio', null, 10000, 50);
						this.describeAColumn(table, 'tassaiscrizioneconf_codice_didprog', 'Codice della didattica programmata', null, 11000, 50);
						this.describeAColumn(table, 'tassaiscrizioneconf_codice_didprogcurr', 'Codice del curriculum', null, 12000, 50);
						this.describeAColumn(table, 'tassaiscrizioneconf_codice_didprogori', 'Codice dell\'orientamento', null, 13000, 50);
						this.describeAColumn(table, 'tassaiscrizioneconf_corsisingoli', 'Corsi singoli', null, 14000, null);
						this.describeAColumn(table, 'costoscontodef_title', 'Costo', null, 15200, 2024);
						this.describeAColumn(table, 'corsostudio_title', 'Denominazione Corso di studi', null, 20100, 1024);
						this.describeAColumn(table, 'corsostudio_annoistituz', 'Anno accademico di istituzione Corso di studi', null, 20600, null);
						this.describeAColumn(table, 'corsostudiokind_title', 'Tipo di corso', null, 21200, 50);
						this.describeAColumn(table, 'didprog_title', 'Denominazione Didattica programmata', null, 22100, 1024);
						this.describeAColumn(table, 'didprog_aa', 'Anno accademico Didattica programmata', null, 22200, 9);
						this.describeAColumn(table, 'sede_title', 'Denominazione Sede Didattica programmata', null, 22320, 1024);
						this.describeAColumn(table, 'didprogcurr_title', 'Curriculum', null, 23200, 256);
						this.describeAColumn(table, 'didprogori_title', 'orientamento', null, 24200, 256);
						this.describeAColumn(table, 'struttura_title', 'Denominazione Dipartimento - Scuola', null, 25100, 1024);
						this.describeAColumn(table, 'strutturakind_title', 'Tipologia Tipo Dipartimento - Scuola', null, 25220, 50);
//$objCalcFieldConfig_default$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["idtassaiscrizioneconf"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			getSorting: function (listType) {
				switch (listType) {
					case "default": {
						return "title asc ";
					}
					//$getSortingin$
				}
				return this.superClass.getSorting(listType);
			}

        });

    window.appMeta.addMeta('tassaiscrizioneconfdefaultview', new meta_tassaiscrizioneconfdefaultview('tassaiscrizioneconfdefaultview'));

	}());
